src/bitset.rs

```diff
@@ -309,6 +312,56 @@ impl<T> Iterator for BitIter<T>
     }
 }
 
+struct AtomicBlock {
+    mask: AtomicUsize,
+    atom: AtomSetOnce<Box<[AtomicUsize; 1 << BITS]>>
+}
+
+impl AtomicBlock {
+    fn new() -> AtomicBlock {
+        AtomicBlock {
+            mask: AtomicUsize::new(0),
+            atom: AtomSetOnce::empty()
+        }
+    }
+    fn add(&self, id: Index) -> bool {
+        if self.atom.is_none() {
+            let v = Box::new(unsafe { std::mem::zeroed() });
```

Comment:
There is a chance that we double-allocate there, if two concurrent `add()` are called. Not a big deal though, assuming it's a rare case.